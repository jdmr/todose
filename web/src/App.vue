<template>
    <div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
        <div v-if="!authenticated.id">
            <h1 class="text-5xl tracking-widest">LOGIN</h1>
            <div class="flex flex-col gap-4">
                <div class="flex flex-col gap-2">
                    <label for="username" class="text-lg tracking-wider">Username</label>
                    <input v-model="authenticated.username" type="text" id="username" name="username"
                        class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent" />
                </div>
                <div class="flex flex-col gap-2">
                    <label for="password" class="text-lg tracking-wider">Password</label>
                    <input v-model="authenticated.password" type="password" id="password" name="password"
                        class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent" />
                </div>
                <button
                    class="bg-blue-500 text-blue-50 p-2 w-full rounded hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                    @click="login">
                    <i class="i-mdi:login h-8 w-8"></i>
                </button>
            </div>

        </div>
        <div v-else>
            <h1 class="text-5xl tracking-widest">TODOS</h1>
            <div class="w-full flex flex-col gap-4">
                <h2 class="text-2xl tracking-wider">Users</h2>
                <div class="flex flex-col gap-2">
                    <div v-for="usr in users" :key="usr.id"
                        class="border p-2 rounded text-lg tracking-wider flex justify-between items-center">
                        <div class="flex items-center gap-2">
                            <button
                                class="p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                                :class="[
            usr.id === selectedUser.id
                ? 'bg-blue-500 text-blue-50'
                : 'bg-gray-300 text-gray-800'
        ]" @click="selectUser(usr)">
                                <i class="i-mdi:check h-8 w-8"></i>
                            </button>
                            <div>{{ usr.name }}</div>
                        </div>
                        <div>
                            <button
                                class="bg-red-500 text-red-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                                @click="deleteUser(usr.id)">
                                <i class="i-mdi:trash-can h-8 w-8"></i>
                            </button>
                        </div>
                    </div>
                </div>
                <div class="flex flex-col gap-1">
                    <label for="name" class="text-lg tracking-wider">Add User</label>
                    <div class="grid grid-cols-[1fr_auto] gap-2">
                        <input v-model="user.name" type="text" id="name" name="name"
                            class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"
                            @keyup.enter="addUser" />
                        <button id="add-user-btn"
                            class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                            @click="addUser">
                            <i class="i-mdi:plus h-8 w-8"></i>
                        </button>
                    </div>
                </div>

                <h2 class="text-2xl tracking-wider">Todos</h2>
                <div class="flex flex-col gap-2">
                    <div v-for="td in todos" :key="td.id"
                        class="border p-2 rounded text-lg tracking-wider flex justify-between items-center">
                        <div class="flex gap-2 items-center">
                            <button @click="toggleStatus(td)"
                                class="p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                                :class="{
            'bg-gray-300 text-gray-500': td.status === 'new',
            'bg-yellow-500 text-yellow-50': td.status === 'started',
            'bg-green-500 text-green-50': td.status === 'done'
        }">
                                <i :class="{
            'i-mdi:checkbox-blank-circle-outline':
                td.status === 'new',
            'i-mdi:clock': td.status === 'started',
            'i-mdi:checkbox-marked-circle': td.status === 'done'
        }" class="h-8 w-8"></i>
                            </button>
                            <div>{{ td.title }}</div>
                        </div>
                        <div>
                            <button
                                class="bg-red-500 text-red-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                                @click="deleteTodo(td.id)">
                                <i class="i-mdi:trash-can h-8 w-8"></i>
                            </button>
                        </div>
                    </div>
                </div>
                <div class="flex flex-col gap-1">
                    <label for="title" class="text-lg tracking-wider">Add Todo</label>
                    <div class="grid grid-cols-[1fr_auto] gap-2">
                        <input v-model="todo.title" type="text" id="title" name="title"
                            class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"
                            @keyup.enter="addTodo" />
                        <button
                            class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"
                            @click="addTodo">
                            <i class="i-mdi:plus h-8 w-8"></i>
                        </button>
                    </div>
                </div>
            </div>
        </div>
    </div>
</template>

<script setup lang="ts">
import { nanoid } from 'nanoid';
import { ref } from 'vue';
import { jwtDecode } from 'jwt-decode';

interface User {
    id: string
    name: string
    username: string
    password: string
}

interface Todo {
    id: string
    title: string
    status: string
    owner: User
}

const users = ref({} as User[])
const user = ref({} as User)
const todos = ref({} as Todo[])
const todo = ref({} as Todo)
const selectedUser = ref({} as User)
const authenticated = ref({} as User)
let token = ''

const login = async () => {
    console.log('logging in')
    try {
        const response = await fetch('/api/v1/login', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(authenticated.value)
        })
        const data = await response.json()
        token = data.token
        const td = jwtDecode(data.token) as User
        authenticated.value = {
            id: td.id,
            name: td.name,
            username: td.username,
            password: ''
        }
        fetchUsers()
        fetchTodos()
        console.log('logged in', authenticated.value)
    } catch (error) {
        console.error('error logging in', error)
        alert('Invalid credentials')
    }
}

const addUser = async () => {
    console.log('adding user')
    user.value.id = nanoid()
    console.log('posting to /api/v1/users')
    const response = await fetch('/api/v1/users', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`
        },
        body: JSON.stringify(user.value)
    })
    const data = await response.json()
    users.value.push(data)
    console.log('finished posting to /api/v1/users', users.value)
    user.value.name = ''
}

const deleteUser = async (id: string) => {
    await fetch(`/api/v1/users/${id}`, {
        method: 'DELETE',
        headers: {
            Authorization: `Bearer ${token}`
        }
    })
    fetchUsers()
}

const selectUser = (usr: User) => {
    console.log(`Selected user: ${usr.name}`)
    selectedUser.value = usr
}

const fetchUsers = async () => {
    console.log('fetching users')
    const response = await fetch('/api/v1/users', {
        headers: {
            Authorization: `Bearer ${token}`
        }
    })
    console.log('assigned users')
    users.value = await response.json()
    console.log('fetched users', users.value)
}

const fetchTodos = async () => {
    const response = await fetch('/api/v1/todos', {
        headers: {
            Authorization: `Bearer ${token}`
        }
    })
    todos.value = await response.json()
}

const addTodo = async () => {
    todo.value.id = nanoid()
    todo.value.status = 'new'
    todo.value.owner = selectedUser.value
    await fetch('/api/v1/todos', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`
        },
        body: JSON.stringify(todo.value)
    })
    todo.value.title = ''
    fetchTodos()
}

const deleteTodo = async (id: string) => {
    await fetch(`/api/v1/todos/${id}`, {
        method: 'DELETE',
        headers: {
            Authorization: `Bearer ${token}`
        }
    })
    fetchTodos()
}

const toggleStatus = async (td: Todo) => {
    console.log(`Toggling status of todo: ${td.title}`)
    switch (td.status) {
        case 'new':
            td.status = 'started'
            break
        case 'started':
            td.status = 'done'
            break
        case 'done':
            td.status = 'new'
            break
    }

    await fetch(`/api/v1/todos/${td.id}`, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`
        },
        body: JSON.stringify(td)
    })
}
</script>
