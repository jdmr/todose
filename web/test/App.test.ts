import { flushPromises, mount } from '@vue/test-utils'
import { afterAll, afterEach, beforeAll, expect, test } from 'vitest'
import App from '../src/App.vue'

import { http, HttpResponse } from 'msw'
import { setupServer } from 'msw/node'

export const restHandlers = [
    http.get('/api/v1/users', ({}) => {
        return HttpResponse.json([
            { id: 'test1', name: 'John Doe' },
            { id: 'test2', name: 'Jane Doe' }
        ])
    }),
    http.post('/api/v1/users', ({}) => {
        return HttpResponse.json(
            { id: 'test3', name: 'New User' }
        )
    }),
    http.put('/api/v1/users/test1', ({}) => {
        return HttpResponse.json(
            { id: 'test1', name: 'Fred Flintstone' }
        )
    }),
    http.delete('/api/v1/users/test1', ({}) => {
        return HttpResponse.json('ok')
    }),
    http.get('/api/v1/todos', ({}) => {
        return HttpResponse.json([
            { id: 'test1', title: 'Todo 1', status: 'new', owner: { id: 'test1', name: 'John Doe'} },
            { id: 'test2', title: 'Todo 2', status: 'new', owner: { id: 'test1', name: 'John Doe'} }
        ])
    }),
    http.post('/api/v1/todos', ({}) => {
        return HttpResponse.json(
            { id: 'test3', title: 'New Todo', status: 'new', owner: { id: 'test1', name: 'John Doe'} }
        )
    }),
    http.put('/api/v1/todos/test1', ({}) => {
        return HttpResponse.json(
            { id: 'test1', title: 'Todo 1', status: 'started', owner: { id: 'test1', name: 'John Doe'} }
        )
    }),
    http.delete('/api/v1/todos/test1', ({}) => {
        return HttpResponse.json('ok')
    })
]
const server = setupServer(...restHandlers)
// Start server before all tests
beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
//  Close server after all tests
afterAll(() => server.close())
// Reset handlers after each test `important for test isolation`
afterEach(() => server.resetHandlers())

test('mount App', async () => {
    expect(App).toBeTruthy()
    const wrapper = mount(App)
    expect(wrapper.text()).toContain('TODOS')
    console.log(wrapper.text())
})

test('fetch users', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
})

test('add user', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#name').setValue('New User')
    await wrapper.find('#add-user-btn').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('New User')
})

test('add user by pressing enter key', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#name').setValue('New User')
    await wrapper.find('#name').trigger('keyup.enter')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('New User')
})

test('select user', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#select-user-test1').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
})

test('delete user', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#delete-user-test1').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).not.toContain('John Doe')
})

test('fetch todos', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
})

test('add todo', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#title').setValue('New Todo')
    await wrapper.find('#add-todo-btn').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('New Todo')
})

test('add todo by pressing enter key', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#title').setValue('New Todo')
    await wrapper.find('#title').trigger('keyup.enter')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('New Todo')
})

test('update todo status', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#toggle-status-test1').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).toContain('Todo 1')
})

test('delete todo', async () => {
    const wrapper = mount(App)
    await flushPromises()
    await flushPromises()
    expect(wrapper.text()).toContain('John Doe')
    expect(wrapper.text()).toContain('Jane Doe')
    expect(wrapper.text()).toContain('Todo 1')
    expect(wrapper.text()).toContain('Todo 2')
    await wrapper.find('#delete-todo-test1').trigger('click')
    await flushPromises()
    await flushPromises()
    console.log(wrapper.text())
    expect(wrapper.text()).not.toContain('Todo 1')
})