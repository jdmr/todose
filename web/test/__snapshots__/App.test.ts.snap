// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`add user 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;

exports[`add user by pressing enter key 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;

exports[`delete user 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;

exports[`fetch users 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;

exports[`mount App 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;

exports[`select user 1`] = `
"<div class="container mx-auto max-w-prose p-2 min-h-screen flex flex-col gap-4 items-center justify-center">
  <h1 class="text-5xl tracking-widest">TODOS</h1>
  <div class="w-full flex flex-col gap-4">
    <h2 class="text-2xl tracking-wider">Users</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="name" class="text-lg tracking-wider">Add User</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="name" name="name" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button id="add-user-btn" class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
    <h2 class="text-2xl tracking-wider">Todos</h2>
    <div class="flex flex-col gap-2"></div>
    <div class="flex flex-col gap-1"><label for="title" class="text-lg tracking-wider">Add Todo</label>
      <div class="grid grid-cols-[1fr_auto] gap-2"><input type="text" id="title" name="title" class="border border-gray-300 rounded p-2 text-xl tracking-wider bg-gray-100 text-gray-800 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent"><button class="bg-blue-500 text-blue-50 p-2 rounded-full hover:brightness-110 hover:shadow-lg focus:brightness-110 focus:shadow-lg transition-all duration-200"><i class="i-mdi:plus h-8 w-8"></i></button></div>
    </div>
  </div>
</div>"
`;
